### PR TITLE
[201911] show ip int changes

### PR DESCRIPTION
**Pull request by arlakshm**
Signed-off-by: Arvindsrinivasan Lakshmi Narasimhan <arlakshm@microsoft.com>

**- What I did**
The PR has changes to support command `show ip interface` for multi asic platform.
Dependent on this PR: https://github.com/Azure/sonic-buildimage/pull/6792

**- How I did it**
 The following changes  are done in this PR
 - The code for `show ip interface` has been moved from `show/main.py` to `scripts\ipintutil`
 -  Modify the code to support both single and multi asic platform
    - To do this the library [pyroute2](https://pypi.org/project/pyroute2/) is used to get interface information in each namespace
   
 - Add the following options for multi asic
     -  [-n, --namespace] to allow user to display the information for given namespaces
         If this option is not present the information from all the namespaces will be displayed

      - [-d, --display] to allow user to display information related both internal and external interfaces
        If this option is not present only external interfaces/neighbors will be display
 - Add unit tests for single and multi asic 

**- How to verify it**
Verify on single and multi asic
```
admin@sonic:~$ show ip interface
Interface        Master    IPv4 address/mask    Admin/Oper    BGP Neighbor    Neighbor IP
---------------  --------  -------------------  ------------  --------------  -------------
Loopback0                  10.1.0.32/32         up/up         N/A             N/A
PortChannel0001            10.0.0.32/31         up/up         ARISTA01T0      10.0.0.33
PortChannel0002            10.0.0.0/31          up/up         ARISTA01T2      10.0.0.1
PortChannel0003            10.0.0.34/31         up/up         ARISTA02T0      10.0.0.35
PortChannel0004            10.0.0.36/31         up/up         ARISTA03T0      10.0.0.37
PortChannel0005            10.0.0.4/31          up/up         ARISTA03T2      10.0.0.5
PortChannel0006            10.0.0.38/31         up/up         ARISTA04T0      10.0.0.39
PortChannel0007            10.0.0.40/31         up/up         ARISTA05T0      10.0.0.41
PortChannel0008            10.0.0.8/31          up/up         ARISTA05T2      10.0.0.9
PortChannel0009            10.0.0.42/31         up/up         ARISTA06T0      10.0.0.43
PortChannel0010            10.0.0.44/31         up/up         ARISTA07T0      10.0.0.45
PortChannel0011            10.0.0.12/31         up/up         ARISTA07T2      10.0.0.13
PortChannel0012            10.0.0.46/31         up/up         ARISTA08T0      10.0.0.47
PortChannel0013            10.0.0.48/31         up/up         ARISTA09T0      10.0.0.49
PortChannel0014            10.0.0.50/31         up/up         ARISTA10T0      10.0.0.51
PortChannel0015            10.0.0.52/31         up/up         ARISTA11T0      10.0.0.53
PortChannel0016            10.0.0.54/31         up/up         ARISTA12T0      10.0.0.55
PortChannel0017            10.0.0.56/31         up/up         ARISTA13T0      10.0.0.57
PortChannel0018            10.0.0.58/31         up/up         ARISTA14T0      10.0.0.59
PortChannel0019            10.0.0.60/31         up/up         ARISTA15T0      10.0.0.61
PortChannel0020            10.0.0.62/31         up/up         ARISTA16T0      10.0.0.63
PortChannel0021            10.0.0.64/31         up/up         ARISTA17T0      10.0.0.65
PortChannel0022            10.0.0.66/31         up/up         ARISTA18T0      10.0.0.67
PortChannel0023            10.0.0.68/31         up/up         ARISTA19T0      10.0.0.69
PortChannel0024            10.0.0.70/31         up/up         ARISTA20T0      10.0.0.71
docker0                    240.127.1.1/24       up/up         N/A             N/A
eth0                       10.3.147.150/23      up/up         N/A             N/A
lo                         127.0.0.1/8          up/up         N/A             N/A
admin@sonic:~$ 


#### Previous command output (if the output of a command-line utility has changed)

#### New command output (if the output of a command-line utility has changed)

### CODE REVIEW

File: sonic-utilities-tests/show_ip_int_test.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    new_output = '\n'.join([line for line in lines if not any(i in line for i in ignored_intfs)])
+    print(new_output)
+    assert new_output == expected_output
+
+
+@pytest.mark.usefixtures('setup_teardown_single_asic')
+class TestShowIpInt(object):
+
+    def test_show_ip_intf_v4(self):
+        return_code, result = get_result_and_return_code(" ipintutil")
```
Comment (jleveque) [inline sonic-utilities-tests/show_ip_int_test.py:104]: Delete space between `"` and `ipintutil`

File: sonic-utilities-tests/show_ip_int_test.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        assert return_code == 0
+        verify_output(result, show_ipv4_intf_with_multple_ips)
+
+    def test_show_ip_intf_v6(self):
+        return_code, result = get_result_and_return_code(" ipintutil -a ipv6")
```
Comment (jleveque) [inline sonic-utilities-tests/show_ip_int_test.py:109]: Delete space between `"` and `ipintutil`

File: sonic-utilities-tests/show_ip_int_test.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+        assert return_code == 0
+        verify_output(result, show_ipv6_intf_with_multiple_ips)
+
+    def test_show_intf_invalid_af_option(self):
+        return_code, result = get_result_and_return_code(" ipintutil -a ipv5")
```
Comment (jleveque) [inline sonic-utilities-tests/show_ip_int_test.py:115]: Delete space between `"` and `ipintutil`

File: sonic-utilities-tests/show_ip_int_test.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        assert return_code == 0
+        verify_output(result, show_multi_asic_ipv6_intf)
+
+    def test_show_ip_intf_v6_all(self):
+        return_code, result = get_result_and_return_code("ipintutil -a ipv6 -d all")
+        assert return_code == 0
+        verify_output(result, show_multi_asic_ipv6_intf_all)
+
+    def test_show_intf_invalid_af_option(self):
+        return_code, result = get_result_and_return_code(" ipintutil -a ipv5")
```
Comment (jleveque) [inline sonic-utilities-tests/show_ip_int_test.py:154]: Delete space between `"` and `ipintutil`

File: sonic-utilities-tests/show_ip_int_test.py [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+        assert return_code == 1
+        assert result == show_error_invalid_af
```
Comment (jleveque) [inline sonic-utilities-tests/show_ip_int_test.py:156]: Add newline at EOF

**Review by jleveque — CHANGES_REQUESTED** [review]
As comments.

Also please fix LGTM alerts

File: sonic-utilities-tests/show_ip_int_test.py (hunk 1 above)
Comment (arlakshm) [reply sonic-utilities-tests/show_ip_int_test.py]: Fixed in latest commit

File: sonic-utilities-tests/show_ip_int_test.py (hunk 3 above)
Comment (arlakshm) [reply sonic-utilities-tests/show_ip_int_test.py]: Fixed in latest commit

File: sonic-utilities-tests/show_ip_int_test.py (hunk 5 above)
Comment (arlakshm) [reply sonic-utilities-tests/show_ip_int_test.py]: Fixed in latest commit

File: sonic-utilities-tests/show_ip_int_test.py (hunk 2 above)
Comment (arlakshm) [reply sonic-utilities-tests/show_ip_int_test.py]: Fixed in latest commit

File: sonic-utilities-tests/show_ip_int_test.py (hunk 4 above)
Comment (arlakshm) [reply sonic-utilities-tests/show_ip_int_test.py]: Fixed in latest commit

**Review by jleveque — CHANGES_REQUESTED** [review]
LGTM alerts still present

**Comment by arlakshm** [comment]
> LGTM alerts still present

Fixed them in latest commit

**Review by jleveque — APPROVED** [review]

**Comment by arlakshm** [comment]
retest this please

**Comment by arlakshm** [comment]
retest this please

**Comment by arlakshm** [comment]
retest this please

**Review by abdosi — APPROVED** [review]

**Pull request merged**